### ISSUE TITLE
Role="main" and <main> interaction with aria-hidden="true" creates false positive

### ISSUE THREAD

**Issue by MatthewSbar** [dequelabs/axe-core#2823]
Expectation: The test "Ensures the document has at most one main landmark" should respect `aria-hidden` when detecting `<main>` elements

Actual: Both `<main>` and `<div role="main">` are counted as main landmarks despite `<main>` being a child of an element with `aria-hidden="true"`

Motivation: This creates a double standard for testing, particularly with modals. If a page normally has a `<main>` element, and opening a modal adds `aria-hidden` to that element's parent, the page is (correctly) seen as having no `main` landmark, however, if a `role="main"` is added within the modal, the `aria-hidden` `<main>` element is counted towards the maximum of one. In other words, the test to see if there is any `main` landmark respects `aria-hidden`, but the test to see if thee is no more than one does not.

Here's an example page that demonstrates the issue:

```
<html lang="en">
  <body>
    <title>Fake page</title>
    <div aria-hidden="true">
      <h1>Hidden page heading</h1>
      <main>hidden main</main>
    </div>
    <div role="dialog" aria-label="fake dialog">
      <h1>Fake dialog heading</h1>
      <div role="main">Fake dialog main</div> <!-- This line is causing a failure -->
    </div>
  </body>
</html>
```
 
In the above example, removing `role-"main"` will mean the page has 0 `main` landmarks, including it as is means the page will have >1 `main` landmarks.
 
<pre><code>
axe-core version: 4.1.2
axe-webdriver, extension or other integration version: 4.7.2

Browser and Assistive Technology versions
</code></pre>

**Comment by straker** [comment]
Good catch.

**Comment by Dragoneric** [comment]
Good finding :-) 
Example: Cookie-Disclaimer on Websites are often used in the shown HTML-syntax

**Comment by WilcoFiers** [comment]
Agreed, this needs to be changed. Thanks for reporting. In addition, when fixing this I think we need to make sure we continue to pass the following:

```html
<html lang="en">
  <body>
    <title>Fake page</title>
    <div aria-hidden="true">
      <h1>Hidden page heading</h1>
      <main>hidden main</main>
    </div>
    <div role="dialog" aria-label="fake dialog">
      <h1>Fake dialog heading</h1>
      <div>Fake dialog main</div> <!-- This line is causing a failure -->
    </div>
  </body>
</html>
```

**Comment by Dragoneric** [comment]
@WilcoFiers  For me this example is wrong, because you have no reachable main tag. It's under aria-hidden=true, so a screenrader can't find a main as strcture element on the page.

**Comment by padmavemulapati** [comment]
Validated with the  latest develop branch axe-core code base, (dated 10th july 2021)
test snippet:
```
<html lang="en">
  <body>
    <title>Fake page</title>
    <div aria-hidden="true">
      <h1>Hidden page heading</h1>
      <main>hidden main</main>
    </div>
    <div role="dialog" aria-label="fake dialog">
      <h1>Fake dialog heading</h1>
      <div>Fake dialog main</div> <!-- This line is causing a failure -->
    </div>
  </body>
</html>
```
Not able to see any violations or best-practices regarding `landmark-no-duplicate-main` / `region` rule

![image](https://user-images.githubusercontent.com/47409579/125153981-213cfb80-e175-11eb-8710-b2edc5b15c05.png)

Seeing a violation on `landmark-no-duplicate-main` with `4.0.2` version
![image](https://user-images.githubusercontent.com/47409579/125154003-5a756b80-e175-11eb-97fa-ca3db6492cda.png)
